src/store/movie/slice.js

```diff
@@ -17,7 +21,12 @@ export const MoviesSlice = createSlice({
   },
   reducers: {
     setMovies(state, action) {
-      state.movies = action.payload;
+      state.page = action.payload;
+    },
+
+    setPaginated(state, action) {
```

Comment:
You can use the same function for both cases and handle the 2 cases (first page and next pages) in this function - i.e. if it's the first page just set the payload in the state, if it's not the first page, concatenate the data to the previous data